src/app/static/src/app/components/ErrorAlert.vue

```diff
@@ -1,11 +1,63 @@
 <template>
     <div class="pt-1 text-danger">
-        {{message}}
+        <div class="error-message">{{message}}</div>
+        <div v-if="hasTrace" >
+            <a href="#" @click="toggleTrace"><strong>{{traceLinkText}}</strong></a>
+            <div v-if="showTrace" class="ml-3">
+                <div v-for="(traceMessage, index) in error.trace" :key="index" class="error-trace">{{traceMessage}}</div>
+            </div>
+        </div>
     </div>
 </template>
 
 <script lang="ts">
-    export default {
-        props: ["message"]
+    import Vue from "vue";
+    import {Error} from "../generated.d";
+
+    interface Data {
+        showTrace: Boolean
+    }
+
+    interface Props {
+        error: Error | null
```

Comment:
This seems a bit odd, so the component might exist even though the error is null? The only place I can spot this happening is https://github.com/mrc-ide/hint/pull/199/files#diff-4c86fe55240bb0dc3f3348916613943dR19 - if we add a `v-if` there then can remove null checking logic from this component. 

Ideally the component would just not render if the error is null. It would be easy with `redux/recompose`! I think there is a functional syntax for vue components, maybe once this project time pressure is over we should look into it.